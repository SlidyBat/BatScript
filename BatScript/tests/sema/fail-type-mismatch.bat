def f():
	if true:
		return 1
	else:
		return true

int g(): return 10

def h(): return

string x = g()
int y = "Hi"
var z = h()