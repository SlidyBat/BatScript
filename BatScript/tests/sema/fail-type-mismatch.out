[sema\fail-type-mismatch.bat:5:2] Error: Return value type (bool) does not match function type (int)
[sema\fail-type-mismatch.bat:11:1] Error: Cannot assign expression of type int to variable of type string
[sema\fail-type-mismatch.bat:12:0] Error: Cannot assign expression of type string to variable of type int
[sema\fail-type-mismatch.bat:13:0] Error: 'void' is an invalid variable type
